controllers/pod.go

```diff
@@ -713,3 +728,272 @@ func getPodNames(pods []*corev1.Pod) []string {
 	}
 	return podNames
 }
+
+func (r *SingleClusterReconciler) handleNSOrDeviceRemoval(rackState RackState) error {
+	podList, err := r.getRackPodList(rackState.Rack.ID)
+	if err != nil {
+		return err
+	}
+	for _, pod := range podList.Items {
```

Comment:
Suggestion: We can do the opposite of what we are doing right now. Instead of calling the `handleNSOrDeviceRemovalPerPod` for each pod, we can identify only once what all the devices and files are removed from ASconfig and then use that info in a podList to do all pod related op